File: cases/taint/go/anonymous-implementation-positive/anonymous_implementation_positive.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: anonymous-implementation-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: anonymous-implementation-sink

type Handler interface {
	Handle(value string)
}

type HandlerFunc func(value string)

func (f HandlerFunc) Handle(value string) {
	f(value)
}

func run() {
	var leakHandler Handler = HandlerFunc(func(value string) { // DFB-WITNESS: anonymous-implementation-handler
		dfb_sink(value)
	})
	var dropHandler Handler = HandlerFunc(func(value string) {
		dfb_sink("clean")
	})
	_ = dropHandler
	leakHandler.Handle(dfb_source())
}

File: cases/taint/rust/anonymous-implementation-positive/anonymous_implementation_positive.rs
fn dfb_source() -> i32 { // DFB-SOURCE: anonymous-implementation-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: anonymous-implementation-sink

fn run() {
    let leak_handler: Box<dyn Fn(i32)> = Box::new(|value| { // DFB-WITNESS: anonymous-implementation-handler
        dfb_sink(value);
    });
    let drop_handler: Box<dyn Fn(i32)> = Box::new(|value| {
        dfb_sink(0);
    });
    leak_handler(dfb_source());
}
